### PR TITLE
rgw: fix memory leaks during Swift Static Website's error handling.

### PR DESCRIPTION
**Pull request by rzarzynski**
Fixes: http://tracker.ceph.com/issues/20757
Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

### CODE REVIEW

**Comment by rzarzynski** [comment]
The patch passed Tempest locally.

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -401,7 +401,8 @@ int rgw_build_bucket_policies(RGWRados* store, struct req_state* s)
   }
 
   if(s->dialect.compare("s3") == 0) {
-    s->bucket_acl = new RGWAccessControlPolicy_S3(s->cct);
+    s->bucket_acl = std::unique_ptr<RGWAccessControlPolicy>(
+        new RGWAccessControlPolicy_S3(s->cct));
```
Comment (cbodley) [inline src/rgw/rgw_op.cc]: consider using `make_unique()` for these (from common/backport14.h)

Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: I'm going to start loving this header. :-) Thanks, @cbodley!

**Comment by rzarzynski** [comment]
jenkins retest this please (due to `test_pidfile.sh` and `osd-dup.sh`)

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-07-26_23:36:55-rgw-wip-yuri-testing_2017_7_27-distro-basic-smithi/

**Pull request merged**